flink-runtime/src/main/java/org/apache/flink/runtime/deployment/InputGateDeploymentDescriptor.java

```diff
@@ -59,7 +71,10 @@
     @Nonnegative private final int consumedSubpartitionIndex;
 
     /** An input channel for each consumed subpartition. */
-    private final ShuffleDescriptor[] inputChannels;
+    private ShuffleDescriptor[] inputChannels;
+
+    /** Serialized value of shuffle descriptors. */
+    private MaybeOffloaded<ShuffleDescriptor[]> serializedInputChannels;
 
     public InputGateDeploymentDescriptor(
```

Comment:
This constructor should now be marked as `VisibleForTesting`. And it should be changed to invoke the production constructor, i.e. 
```
this(
  consumedResultId, 
  consumedPartitionType, 
  consumedSubpartitionIndex, 
  new NonOffloaded(new SerializedValue(inputChannels)));
```

Reply:
Sorry for missing this. I treat the `StreamNetworkBenchmarkEnvironment` as a production class. Resolved.